### PR TITLE
ci: Build release package with correct telemetry key

### PR DESCRIPTION
**Pull request by grianbrcom**
The telemetry key was injected before the package build.
And wiped out after to not be present in repo.

Signed-off-by: Anton Grigorev <anton.grigorev@broadcom.com>

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -15,7 +15,8 @@ jobs:
       with:
         java-version: 11
     - name: Build with Maven
-      run: cd server; mvn package -DskipTests --no-transfer-progress
+      working-directory: server
+      run: mvn package -DskipTests --no-transfer-progress
```
Comment (temanbrcom) [inline .github/workflows/release.yml:19]: Do we really need to skip tests on release?

**Review by temanbrcom — APPROVED** [review]

File: release/release-it-release.json [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
   "hooks": {
-    "before:init": ["git config user.email \"dont@write.here\"", "git config user.name \"Caring bot\""],
-    "after:bump": "git add ../../CHANGELOG.md; npx vsce package"
+    "before:init": [
+      "git config user.email \"dont@write.here\"",
+      "git config user.name \"Caring bot\""
+    ],
+    "after:bump": [
+      "git add ../../CHANGELOG.md",
+      "echo $THE_KEY > resources/TELEMETRY_KEY",
```
Comment (filipkroupa) [inline release/release-it-release.json:9]: Would the env variable $THE_KEY be visible in log here? I guess only secrets are obfuscated

File: .github/workflows/release.yml (hunk 1 above)
Comment (grianbrcom) [reply .github/workflows/release.yml:19]: They fail, and I don't know why. I can check the current state.

File: release/release-it-release.json (hunk 2 above)
Comment (grianbrcom) [reply release/release-it-release.json:9]: The value is not present in logs.
You can see an example here: https://github.com/grianbrcom/che-che4z-lsp-for-cobol/runs/1442316084?check_suite_focus=true in "Make a release" section.

Comment (filipkroupa) [reply release/release-it-release.json:9]: Nice, thanks

**Review by filipkroupa — APPROVED** [review]

**Pull request merged**